src/main/java/de/tum/in/www1/artemis/service/connectors/aeolus/AeolusTemplateService.java

```diff
@@ -44,6 +44,35 @@ public class AeolusTemplateService {
     public AeolusTemplateService(ProgrammingLanguageConfiguration programmingLanguageConfiguration, ResourceLoaderService resourceLoaderService) {
         this.programmingLanguageConfiguration = programmingLanguageConfiguration;
         this.resourceLoaderService = resourceLoaderService;
+        // load all scripts into the cache
+        cacheOnBoot();
+    }
+
+    private void cacheOnBoot() {
+        var resources = this.resourceLoaderService.getResources(Path.of("templates", "aeolus"));
+        for (var resource : resources) {
+            try {
+                String filename = resource.getFilename();
+                if (filename == null) {
+                    continue;
+                }
+                String directory = resource.getURL().getPath().split("templates/aeolus/")[1].split("/")[0];
+                String projectType = filename.split("_")[0].replace(".yaml", "");
+                Optional<ProjectType> optionalProjectType = Optional.empty();
+                if (!projectType.equals("default")) {
+                    optionalProjectType = Optional.of(ProjectType.valueOf(projectType.toUpperCase()));
+                }
+                String uniqueKey = directory + "_" + filename;
+                byte[] fileContent = IOUtils.toByteArray(resource.getInputStream());
+                String script = new String(fileContent, StandardCharsets.UTF_8);
+                Windfile windfile = readWindfile(script);
+                this.addInstanceVariablesToWindfile(windfile, ProgrammingLanguage.valueOf(directory.toUpperCase()), optionalProjectType);
+                templateCache.put(uniqueKey, windfile);
+            }
+            catch (IOException | IllegalArgumentException e) {
+                logger.error("Failed to load windfile {}", resource.getFilename(), e);
+            }
```

Comment:
Question: If the loading of Windfiles was flaky before, is it a good idea to just ignore this error here? What effect would that have on other code that needs these templates later on?